quinn-proto/src/connection/streams/state.rs

```diff
@@ -782,7 +801,10 @@ impl StreamsState {
                     StreamHalf::Recv => !self.send.contains_key(&id),
                 };
             if fully_free {
-                self.alloc_remote_stream(id.dir());
+                self.remote_count[id.dir() as usize] -= 1;
+                if self.remote_count[id.dir() as usize] < self.max_remote_count[id.dir() as usize] {
```

Comment:
Did you consider putting this condition inside `alloc_remote_streams()`? I think that would prevent the repetition here and in `set_max_concurrent()`?

Reply:
Tried this out in a separate commit. It's a semantic change, but I can see how it might be nicer. Thoughts?

Reply:
It does seem to centralize the logic more which seems nice? You decide which you prefer.